core/src/autogluon/core/trainer/abstract_trainer.py

```diff
@@ -1689,70 +1729,6 @@ def refit_single_full(
-            if not ray.is_initialized():
-                ray.init(log_to_driver=False, logging_level=logging.ERROR)
-
-            distributed_manager = ParallelFitManager(
-                mode="refit",
-                func=_remote_refit_single_full,
-                func_kwargs=dict(fit_strategy=fit_strategy),
-                func_put_kwargs=dict(
-                    _self=self,
-                    X=X,
-                    y=y,
-                    X_val=X_val,
-                    y_val=y_val,
-                    X_unlabeled=X_unlabeled,
-                    kwargs=kwargs,
-                ),
-                # TODO: check if this is available in the kwargs
-                num_cpus=kwargs.get("total_resources", {}).get("num_cpus", 1),
-                num_gpus=kwargs.get("total_resources", {}).get("num_gpus", 0),
-                get_model_attribute_func=self.get_model_attribute,
-                X=X,
-                y=y,
-            )
-
-            for level in levels:
-                models_trained_full_level = []
-                distributed_manager.job_kwargs["level"] = level
-                models_level = model_levels[level]
-
-                logger.log(20, f"Scheduling distributed model-workers for refitting {len(models_level)} L{level} models...")
-                unfinished_job_refs = distributed_manager.schedule_jobs(models_to_fit=models_level)
-
-                while unfinished_job_refs:
-                    finished, unfinished_job_refs = ray.wait(unfinished_job_refs, num_returns=1)
-                    refit_full_parent, model_trained, model_path, model_type = ray.get(finished[0])
-
-                    self._add_model(
-                        model_type.load(path=os.path.join(self.path,model_path), reset_paths=self.reset_paths),
-                        stack_name=REFIT_FULL_NAME,
-                        level=level,
-                        _is_refit=True
-                    )
-                    model_refit_map[refit_full_parent] = model_trained
-                    self._update_model_attr(
-                        model_trained,
-                        refit_full=True,
-                        refit_full_parent=refit_full_parent,
-                        refit_full_parent_val_score=self.get_model_attribute(refit_full_parent,"val_score"),
-                    )
-                    models_trained_full_level.append(model_trained)
-
-                    logger.log(20,f"Finished refit model for {refit_full_parent}")
-                    unfinished_job_refs += distributed_manager.schedule_jobs()
-
-                logger.log(20, f"Finished distributed refitting for {len(models_trained_full_level)} L{level} models.")
-                models_trained_full += models_trained_full_level
-                distributed_manager.clean_job_state(unfinished_job_refs=unfinished_job_refs)
-
-            distributed_manager.clean_up_ray()
```

Comment:
Why are we deleting this code?

Reply:
It's dead code (i.e., is never used)

Reply:
cf. below. type checker warned code is unreachable, up to you if you'd like to keep it.

https://github.com/autogluon/autogluon/blob/bd24491f6c7e0ea4468644d6781573c578a73d4a/core/src/autogluon/core/trainer/abstract_trainer.py#L1675-L1677

Reply:
I agree it is dead code, but I would like to keep it since this was a stretch goal for 1.2 release that didn't quite make it since I didn't have time to comprehensively test it. The plan is it will be used in 1.3 release, since it is a major speedup, but it needs to be made memory safe (current implementation isn't necessarily memory safe).

Maybe instead of removing you could add a FIXME / TODO at the top of the code mentioning that this is currently dead code.

Reply:
Basically, I don't want to forget that a lot of work towards this feature has already been done, which could very well happen if we delete the code.

Reply:
the WIP follow-up PR that uses this code: https://github.com/autogluon/autogluon/pull/4680